flowpipe/plug.py

```diff
@@ -115,7 +118,67 @@ def promote_to_graph(self, name=None):
         self.node.graph.add_plug(self, name=name)
 
 
-class OutputPlug(IPlug):
+class _OutPlug(IPlug):
```

Comment:
We should have a better name for this. `_AutoConnectOutputPlug` or something like that

Reply:
I agree that it's not a perfect name. It does more than just the automatic connection though. All connection logic now resides here. An alternative to this approach is to re-write `SubOutputPlug` to inherit from `OutputPlug` and forbidding the nesting of subplugs (if we want to forbid it). Do you think that would be possible, or is there something inherent in the structure of the Sub-plugs that will stop us from doing so?

Reply:
I think we can do that, subplugs should never be nested any further.